src/QsCompiler/QirGeneration/Interop.cs

```diff
@@ -61,63 +62,69 @@ internal sealed class Interop
             where TOut : class =>
             items.Select(map).Where(i => i != null).Select(i => i!).ToArray();
 
-        /// <inheritdoc cref="MapToInteropType(ITypeRef)"/>
+        /// <inheritdoc cref="MapToInteropType(Context, ITypeRef)"/>
         private ITypeRef? MapToInteropType(ResolvedType type) =>
-            this.MapToInteropType(this.sharedState.LlvmTypeFromQsharpType(type));
+            MapToInteropType(this.sharedState.Context, this.sharedState.LlvmTypeFromQsharpType(type));
+
+        /// <inheritdoc cref="MapToInteropType(Context, ITypeRef)"/>
+        private ITypeRef? MapToInteropType(ITypeRef type) =>
+            MapToInteropType(this.sharedState.Context, type);
 
         /// <summary>
-        /// Maps the given Q#/QIR type to a more interop-friendly type.
-        /// Returns null only if the given type is Unit. Strips items of type Unit inside tuples.
+        /// Maps the given type to an interop-friendly type.
+        /// Strips items of type unit inside tuples.
+        /// Returns null only if the given type is unit, or a tuple of unit values.
         /// </summary>
         /// <exception cref="ArgumentException">
         /// The given type is a pointer to a non-struct type,
         /// or the given type is not specified in the QIR format.
         /// </exception>
-        private ITypeRef? MapToInteropType(ITypeRef t)
+        internal static ITypeRef? MapToInteropType(Context context, ITypeRef t)
```

Comment:
I was trying to avoid having a Context object. I don't have one in the place where I need to call this.

Reply:
You can see where I need this here: https://github.com/microsoft/qsharp-compiler/pull/1014/files#diff-ca1be728ef0634194297443f83770d8360446808f72afe59669d9cdf4bf83c7eR53

Reply:
This is the context of the llvm bindings. You would construct it just with Context(). This is the piece that may require you to have a reference to the llvmlibs. The way around it is to have the entry point object construction not work with the Q# data structures at all but instead construct everything relying on just the information in the QIR itself (which is the better setup for other reasons/people as well).